CMakeLists.txt

```diff
@@ -111,10 +115,29 @@ if(MSVC)
     endif(BOOST_ALL_DYN_LINK)
 endif(MSVC)
 
-include(VolkBoost)
+# Check if we have std::filesystem
+if(POLICY CMP0057)
+    cmake_policy(SET CMP0057 NEW)
+endif()
+
+if(CMAKE_VERSION VERSION_LESS 3.8)
+    set(FILESYSTEM_FOUND FALSE)
+else()
+    find_package(FILESYSTEM COMPONENTS Final Experimental)
+endif()
 
-if(NOT Boost_FOUND)
-    message(FATAL_ERROR "VOLK Requires boost to build")
+#### Set C++ standard to 17 if std::filesystem is available
+if(${FILESYSTEM_FOUND})
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
You also set this value in `FindFILESYSTEM.cmake`. Is this required in both places?

Reply:
I'm not sure, since the CMake module does:
```
include(CMakePushCheckState)
cmake_push_check_state()
...
cmake_pop_check_state()
```
so I'm not sure if the right compiler flags survive. This is why I included that line again.

Reply:
In this case it should be in.